architecture/azure/processes.adoc

```diff
@@ -0,0 +1,63 @@
+[[architecture-azure-processes]]
+= Processes
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+This section presents a summary of how Azure Red Hat OpenShift specific
+processes work. These descriptions do not cover all edge cases and are
+limited to presenting the expected flow. Edge case handling can be observed
+in the link:https://github.com/openshift/openshift-azure[source code repository]
```

Comment:
@Makdaam The OCP docs typically [don't allow links to GitHub repos](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#links-to-external-websites). Have discussed including this link with anyone, yet? 

If not, we might have to migrate relevant content from that repo into this one.

Reply:
I haven't discussed it, I wasn't aware of this policy.
If it's not common the whole sentence about the Edge cases can be changed to "We omit edge case descriptions in the documentation to retain clarity.".
I just wanted to signal that there is some logic which is not described in the docs, like "what if an upgrade fails in the middle" or "what if a master VM loses its hard drive contents".

Reply:
@Makdaam Yeah, it can make things complicated sometimes. :) 

I'm fine with a note explaining that the topic doesn't cover all cases. If there's non-GitHub docs we can link to that might give people an idea of where to look when they're not on the happy path, all the better.